plugins/importexport/native/filter/NativeXmlPKPPublicationFilter.php

```diff
@@ -80,11 +80,12 @@ public function handleElement($node)
         $publication->setData('seq', $node->getAttribute('seq'));
         $publication->setData('accessStatus', $node->getAttribute('access_status'));
         $publication->setData('status', $node->getAttribute('status'));
-        $publication->setData('primaryContactId', $node->getAttribute('primary_contact_id'));
         $publication->setData('urlPath', $node->getAttribute('url_path'));
 
         $publicationId = Repo::publication()->dao->insert($publication);
         $publication = Repo::publication()->get($publicationId);
+        // Non-persisted temporary ID, will be updated and stored once the authors get parsed
+        $publication->setData('primaryContactId', $node->getAttribute('primary_contact_id'));
```

Comment:
This isn't very safe (e.g. in case another part of the code decides to use the `$deployment->setPublication()`), but instead of creating/spreading extra workarounds, maybe it's better to proceed with the refactoring or retirement (JATS?!) of this plugin.
If you want, I can come up with a better approach (store the data in a class property).

The same applies to the other fix.